### ISSUE TITLE
UX improvements to the area detector viewer

### ISSUE THREAD

**Issue by canismarko** [spc-group/haven#232]
During initial commissioning, we used the area_detector.ui window for BPM camera. Sometimes this camera disconnects, but there is no obvious indication that this has happened.

We should at least add labels for the following:

- number of frames collected
- detector state

**Comment by yannachen** [comment]
No minimum/maximum values for the slider in the previous version. I will update in a new pull request.

**Comment by yannachen** [comment]
The slider is still not working after changing the PyDMslider channel to "ca://25idSimDet:Over1:1:CenterX." Temporarily, we set the sliders' limits from 0 to 3000 to make slider working.

**Comment by canismarko** [comment]
> Temporarily, we set the sliders' limits from 0 to 3000 to make slider working.

Yes, but that creates its own UX issues when the range doesn't match the image size. How about we find a time to dig into the PyDM code and see what's going on?

**Comment by canismarko** [comment]
I think we're ready to move forward on this. I created an Aravis-based area detector device in ophyd-async: https://github.com/spc-group/haven/blob/main/src/haven/devices/detectors/aravis.py

I think we can now add back in controls for the rest of the plugins that we need, but those plugins aren't part of the device yet, so need to add them. Here are the steps that I see:

1) Identify which plugins we need widgets for (e.g. overlays)
2) Add those plugins as devices to the main AravisDetector device
3) Update the UI widgets to connect with those new signals
4) Make sure it is easy for users to interact with the UI

* I don't think we need to do ROI or stats anymore since the group decided it was better to do ROI handling as part of the analysis rather than data acquisition.

**Comment by yannachen** [comment]
1. Shape width was added in overlays;
2. AravisDetector has connection status, exposure, gain and image saving size;
3. For exposure/gain, once Continous checkbox is checked, Auto is disabled; Auto button can be clicked after unchecking Continous;
4. Pytest has passed; Testing Firely-Dev from account yannachen shows well except for choosing the overlay shape and shape size/center. Need to see in Firefly.

**Comment by yannachen** [comment]
Pull request has 3 errors, but not from this task. I am not sure how to deal with it.

**Comment by canismarko** [comment]
> Pull request has 3 errors, but not from this task. I am not sure how to deal with it.

Okay, I'll take a look and respond [in the PR](https://github.com/spc-group/haven/pull/354)

**Comment by canismarko** [comment]
Looks like there was a [merge commit](https://github.com/spc-group/haven/commit/273f7d86f5175373db998bf1976e56d942d527ba) to bring your branch up to date with the spc-group repo (or something like that). During that merge, looks like there was a conflict that was not resolved. If git can't figure out how to merge things, it prints a message saying there are conflicts, and telling you they need to be resolved before the merge can be completed. It also puts markers into to file so you can see what the conflict is. In this case, it looks like you completed the merge without actually resolving the conflicts so the markers were left in. Since those markers are not valid XML, Qt cannot open the resulting UI files. Take a look at "Files Changed" tab and find the line_scan.ui file. You'll find a "<<<" and similar are being added by this PR. You need to resolve that conflict.

This might be helpful: https://www.geeksforgeeks.org/merge-conflicts-and-how-to-handle-them/#handling-the-merge-conflict
